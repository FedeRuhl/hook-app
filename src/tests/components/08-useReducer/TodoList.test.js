import { shallow } from "enzyme";
import TodoList from "../../../components/08-useReducer/TodoList"
import demoTodos from "../../fixtures/demoTodos";

describe('TodoList tests', () => {

    const handleToggle = jest.fn();
    const handleDelete = jest.fn();

    const wrapper = shallow(
        <TodoList
            todos = { demoTodos }
            handleToggle = { handleToggle }
            handleDelete = { handleDelete }
        />
    );

    test('should show the component succesfully', () => {
        expect(wrapper).toMatchSnapshot();  
    });

    test('should has 2 items', () => {
       expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length); 
       expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function)); 
       expect(wrapper.find('TodoListItem').at(0).prop('handleToggle')).toEqual(expect.any(Function)); 
    });
    
    
});