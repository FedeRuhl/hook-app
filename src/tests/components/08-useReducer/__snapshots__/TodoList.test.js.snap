// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList tests should show the component succesfully 1`] = `
<ul
  className="list-group"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "learn react",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "learn flutter",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
